### PR TITLE
stub: stabilize StreamObserver APIs

### PR DESCRIPTION
**Pull request by dapengzhang0**
Resolving #1788

### CODE REVIEW

File: stub/src/main/java/io/grpc/stub/CallStreamObserver.java [hunk 1]
```diff
@@ -50,7 +48,6 @@
  * <p>DO NOT MOCK: The API is too complex to reliably mock. Use InProcessChannelBuilder to create
  * "real" RPCs suitable for testing.
  */
-@ExperimentalApi("https://github.com/grpc/grpc-java/issues/1788")
 public abstract class CallStreamObserver<V> implements StreamObserver<V> {
```
Comment (ejona86) [inline stub/src/main/java/io/grpc/stub/CallStreamObserver.java:51]: I thought this class was to remain experimental.

Comment (dapengzhang0) [reply stub/src/main/java/io/grpc/stub/CallStreamObserver.java:51]: If this class remains experimental, then its subclasses should also be experimental. We can not have a stable class extend an experimental class.

Comment (ejona86) [reply stub/src/main/java/io/grpc/stub/CallStreamObserver.java:51]: We talked of redefining the methods in the subclasses, so the only issue is if someone references CallStreamObserver directly.

Comment (dapengzhang0) [reply stub/src/main/java/io/grpc/stub/CallStreamObserver.java]: Reverted `CallStreamObserver.java`.

Comment (dapengzhang0) [reply stub/src/main/java/io/grpc/stub/CallStreamObserver.java]: Actually I don't recall the details and don't find anything specific in the experimental API meeting notes or the issue #1788 . @ejona86 What else is needed?

Comment (ejona86) [reply stub/src/main/java/io/grpc/stub/CallStreamObserver.java]: I was referencing https://github.com/grpc/grpc-java/issues/1788#issuecomment-453227259 before.

Comment (dapengzhang0) [reply stub/src/main/java/io/grpc/stub/CallStreamObserver.java]: Redefined methods of `CallStreamObserver` in subclasses.

**Pull request merged**